### PR TITLE
dird: fix config reload and unit tests dependency issue

### PR DESCRIPTION
**Pull request by alaaeddineelamri**
#### Description

When creating google tests that include `ua_cmd`, a dependency issue arises around the `DoReloadConfig()` function and forces developers to re-implement it within the test to return a random `false`.  This PR fixes the issue.

Waiting on PR #1151 , and then need to be redone

#### Please check

- [x] Short description and the purpose of this PR is present _above this paragraph_
- [x] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)

##### General
- [x] PR name is meaningful
- [x] Purpose of the PR is understood
- [x] Separate commit for this PR in the CHANGELOG.md, PR number referenced is same
- [x] Commit descriptions are understandable and well formatted

##### Source code quality

- [x] Source code changes are understandable
- [x] Variable and function names are meaningful
- [x] Code comments are correct (logically and spelling)
- [x] Required documentation changes are present and part of the PR
- [x] `bareos-check-sources --since-merge` does not report any problems
- [x] `git status` should not report modifications in the source tree after building and testing

### CODE REVIEW

File: core/src/dird/reload.cc [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+
+/*
+ * See if two storage definitions point to the same Storage Daemon.
+ *
+ * We compare:
+ *  - address
+ *  - SDport
+ *  - password
+ */
+static inline bool IsSameStorageDaemon(StorageResource* store1,
```
Comment (pstorz) [inline core/src/dird/reload.cc:36]: This function also exists in migrate.cc. Can it be extracted?

**Review by pstorz — CHANGES_REQUESTED** [review]
Extrating the IsSameStorageDaemon() would be great.
Other than that: Good work!

File: core/src/dird/reload.cc (hunk 1 above)
Comment (alaaeddineelamri) [reply core/src/dird/reload.cc:36]: yes it can

**Review by pstorz — APPROVED** [review]
Good work. As soon as it successfully builds we can merge.

**Review by pstorz — APPROVED** [review]
Good work!

**Pull request merged**